coala_quickstart/generation/Project.py

```diff
@@ -25,3 +26,66 @@ def get_project_dir(printer):
             printer.print("Please enter a valid directory", color="blue")
 
     return project_dir
+
+
+def get_popular_languages(file_paths):
+    """
+    Identifies the most popular languages in the user's project directory
+    from the files matched from the given glob expression.
+
+    :param file_paths:
+        A list of absolute file paths in the user's project directory.
+    :return:
+        A list of tuples containing a language name as the first value
+        and percentage usage in the project as the second value.
+    """
+    results = {}
+    for ext in exts:
+        results[exts[ext]] = 0
+    results["Unknown"] = 0
+
+    for file_path in file_paths:
+        name, ext = os.path.splitext(file_path)
+        if ext in exts:
+            results[exts[ext]] += 1
+        else:
+            results["Unknown"] += 1
+    if len(file_paths):
+        for ext in results:
+            results[ext] = (100.0 * results[ext]) / len(file_paths)
+
+    languages = sorted(
+        results.items(),
+        key=operator.itemgetter(1),
+        reverse=True)
+    used_languages = []
+    append_list = []
+    for lang, percent in languages:
+        if lang == "Unknown":
+            append_list += [(lang, int(percent))]
+            continue
+        if percent == 0:
+            break
+        else:
+            used_languages += [(lang, int(percent))]
+
+    return used_languages + append_list
+
+
+def print_popular_languages(printer, results):
+    """
+    Prints the sorted list of popular languages along with each language's
+    percentage use.
+
+    :param printer:
+        A ``ConsolePrinter`` object used for console interactions.
+    :param results:
+        A list of tuples containing a language name as the first value
+        and percentage usage in the project as the second value.
+    """
+    if len(results):
+        print("\nThe following languages have been automatically detected:")
```

Comment:
Preceeding `\n` because the previous line is user inputted line so I think a linebreak would look good there.